nixos/modules/config/krb5/default.nix

```
@@ -286,13 +289,82 @@ in {
         '';
       };
     };
+
+    security.pam =
+      let
+        pamModCfg = pamCfg.modules.${pamModName};
+        path = "${pkgs.pam_krb5}/lib/security/pam_krb5.so";
+        ccreds_path = "${pkgs.pam_ccreds}/lib/security/pam_ccreds.so";
+      in
+      utils.pam.mkPamModule {
+        name = pamModName;
+        mkSvcConfigCondition = svcCfg: cfg.enable && svcCfg.modules.${pamModName}.enable;
+
+        mkModuleOptions = global: {
+          enable = mkOption {
+            type = types.bool;
+            default = if global then true else pamModCfg.enable;
```

```suggestion
        mkSvcConfigCondition = svcCfg: svcCfg.modules.${pamModName}.enable;

        mkModuleOptions = global: {
          enable = mkOption {
            type = types.bool;
            default = if global then cfg.enable else pamModCfg.enable;
```
Should we do this instead?